crates/relayer/src/supervisor.rs

```diff
@@ -280,6 +283,18 @@ pub fn spawn_rest_worker<Chain: ChainHandle>(
     )
 }
 
+/// Spawn a background task which verifies if there are idle workers and removes them if.
+pub fn spawn_cleanup_worker(workers: Arc<RwLock<WorkerMap>>) -> TaskHandle {
+    spawn_background_task(
+        error_span!("cleanup_worker"),
+        Some(Duration::from_secs(300)),
```

Comment:
That sounds a bit excessive. Let's do it once per second instead.

Reply:
Ok, this should also drastically reduce how long the test runs